pennylane/tape/cv_param_shift.py

```diff
@@ -370,9 +370,9 @@ def processing_fn(results):
                 array[float]: 1-dimensional array of length determined by the tape output
                 measurement statistics
             """
-            res = results[0]
+            res = np.array(results)[0]
             grad = np.zeros_like(res)
-            grad[transformed_obs_idx] = res

```

Comment:
The previous implementation assumed that `res` was always a scalar. This is only the case if the QNode has only one observable.